### PR TITLE
[Functions] Extract only allowed fields from CBOR payload

### PR DESCRIPTION
**Pull request by bolekk**
We used to translate CBOR to JSON and send it to the EA without any validations.
With this change, we first translate CBOR into a RequestData struct, which contains all required fields with correct types. Only then, we marshal it to JSON and send to the EA.
That way we:
- catch all parsing errors early (e.g. incorrect types)
- don't put load on the EA unnecessarily if the data is invalid or very big
- make it easier to manage fields on the node's side (e.g. secrets or metadata) in the future.

### CODE REVIEW

**Review by KuphJr — COMMENTED** [review]
I have some questions about the impact of switching to using the `ParseDietCBORToStruct` function.  Will this have any actual change in behavior?  What does `autoAddMapDelimiters` actually do?

File: core/cbor/cbor.go [hunk 1]
```diff
@@ -34,6 +34,13 @@ func ParseDietCBOR(b []byte) (map[string]interface{}, error) {
 	return output, nil
 }
 
+// Similar to ParseDietCBOR but outputs to a concrete struct, which meets the
+// "top-level map" requirement of "diet" CBOR.
+func ParseDietCBORToStruct(b []byte, v interface{}) error {
```
[the remark this answers is not in the document]
Comment (bolekk) [reply core/cbor/cbor.go:39]: See new tests in listener_test.go. Additionally, contract-side encoding is covered by integration tests.

[the remark this answers is not in the document]
Comment (bolekk) [reply core/cbor/cbor.go:39]: I'll add one to cover invalid field type.

**Comment by bolekk** [comment]
@KuphJr I added more tests and wrote a PR summary to answer your questions.

If you look inside ParseDietCBOR (which was used before) you'll see that it's also autoAddMapDelimiters() and later transforming into a string->iface map. The autoAddMapDelimiters() call adds missing bytes that mark the "map" type in CBOR. Those delimiters are missing on purpose in the "diet" version - everything is explained in the comments in cbor.go if you're curious.

There's no behavior change except that the EA now won't get any fields that should not be part of request data OR are of incorrect type.

**Review by KuphJr — APPROVED** [review]

**Pull request merged**